Feat/comments reply edit delete

fmSand opened the pull request:
- Added reply and edit forms
 - Some more user-friendly error messages
 - Changed 'span' elements to 'time' elements
-  Ran format with prettier so a lot of the "changes" is just that.

*Later PR's*:
 - Pagination
 - Proper ARIA, and better loading states
 - Focus improvements 
 - Improved error handling
 - Enhanced validation
 - Add Icons
 - CSS

fmSand committed:
feat: add reply, edit delete functionality
 -implement reply and edit forms
 -add more userfriendly error messages
 -enhance comment state management
 - remove maxlength validation for temp validation fix
 - change span el to time elements

fmSand committed:
fix: run formatter

fmSand committed:
fix: update deleted comment message for consistency

Anclagen commented on src/components/features/comments/forms.js:
```diff
@@ -1,14 +1,22 @@
-      rows: 4,
-    maxlength: 2000,
+    placeholder: "Write a condolence...",
+    rows: 4,
```
I assume the max length remove is to allow purely js validation

fmSand replied:
When testing >2000 chars, the browser I used would still let me submit, just automatically slicing the comment when displayed. So I removed it to see if the js validation, message and disabled submit button worked without it.  It's added back now. Might add a char counter later.

Anclagen commented on src/components/features/comments/forms.js:
```diff
@@ -70,10 +77,12 @@ export function createMainCommentForm({ onSubmit }) {
       } else {
-        errorElem.textContent = result?.error || "Sorry, we couldn't post your condolence. Please try again.";
+        errorElem.textContent =
+          result?.error ||
```
Wouldn't this be result.message? and if its a validation failure, you would still get a message, but also an array on errors with the field and message for the specific field.

fmSand replied:
Correct. Fixed now

Anclagen commented on src/components/features/comments/forms.js:
```diff
@@ -86,4 +95,206 @@ export function createMainCommentForm({ onSubmit }) {
+    textarea.setSelectionRange(len, len);
+  }, 0);
+  return form;
+}
```
This section contains a lot of repeated logic for form creation and handling, which could be improved by breaking it down into reusable components or utilities. We could streamline this to use some factory function, like:
```js
function createCommentForm({ label, onSubmit, onCancel, initialValue, submitText }) {
  // ... shared form creation logic
}

// functions handleReply, closeReply, etc...

const replyForm = createCommentForm({
  label: "Reply",
  onSubmit: handleReply,
  onCancel: closeReply,
  submitText: "Reply",
});

const editForm = createCommentForm({
  label: "Edit comment",
  onSubmit: handleEdit,
  onCancel: cancelEdit,
  initialValue: comment.text,
  submitText: "Save",
});
```

This would make the forms a bit easier to maintain.

fmSand replied:
Wanted to have a go at this. Was a bit more complicated with some of the differences to the way they behave, so might not have improved much, but I think it turned out ok.

Anclagen reviewed (changes requested):
Looks good. Just update the error message for now, then this can be merged.

The form logic has a lot of repeated code that could be refactored later into smaller pieces. I'm not going to require it for this PR as I just want this finished.

For frontend work, it’s usually cleaner to merge HTML + CSS scaffolding first, then layer in JS functionality. That way you can test layout incrementally without relying on the browser network tab.

fmSand committed:
fix:
- add aria attr to showFieldError
- error fix,
- simplefy reply sync,
- add form factory
- use html for main form.

fmSand committed:
fix(comments): add try/catch to main form submit and require res.data in edit handler

- Wrapped main comment form onSubmit in try/catch to handle network errors and always return a shaped { success, message } object
- Updated edit handler to require both res.success && res.data before applying
  DOM/state updates, preventing optimistic updates with missing server data

Anclagen reviewed (approved):
LGTM, keep up the good work
